### PR TITLE
Empty container loss fix

### PR DESCRIPTION
**Pull request by josephdviviano**
This should handle all edge cases where an empty container (either pre or post "valid" filtering) is handled properly (returns a loss of 0).

### CODE REVIEW

File: src/gfn/gflownet/detailed_balance.py [hunk 1]
```diff
@@ -58,6 +61,10 @@ def __init__(
         self.logF = logF
         self.forward_looking = forward_looking
         self.log_reward_clip_min = log_reward_clip_min
+        if safe_log_prob_min:
+            self.log_prob_min = -1e-10
```
[the remark this answers is not in the document]
Comment (josephdviviano) [reply src/gfn/gflownet/detailed_balance.py:65]: I wasn't sure if the -inf would cause weird issues in unforseen elements of the code.

[the remark this answers is not in the document]
Comment (josephdviviano) [reply src/gfn/gflownet/detailed_balance.py:65]:
```
>>> -1e10
-10000000000.0
```

[the remark this answers is not in the document]
Comment (josephdviviano) [reply src/gfn/gflownet/detailed_balance.py:65]: it's very likely overkill:

```
>>> torch.log(torch.tensor(0.000000000000000000000000000000000000000000001))
tensor(-103.2789)
```

and adding another 0 gives us `-inf`.

[the remark this answers is not in the document]
Comment (hyeok9855) [reply src/gfn/gflownet/detailed_balance.py:65]: What about giving log prob 0 (i.e., p=1) for the empty transition?

**Review by younik — APPROVED** [review]
after solving comments on safe_log_prob_min, looks good

File: src/gfn/gflownet/detailed_balance.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (saleml) [reply src/gfn/gflownet/detailed_balance.py:65]: Hi @josephdviviano 
I still think it should be -1e10 rather than -1e-10. We want probabilities to be close to 0, right, well the log prob should be -> -inf
@hyeok9855 , thoughts?

[the remark this answers is not in the document]
Comment (josephdviviano) [reply src/gfn/gflownet/detailed_balance.py:65]: @saleml do I misunderstand something basic here? These are log probs.

[the remark this answers is not in the document]
Comment (josephdviviano) [reply src/gfn/gflownet/detailed_balance.py:65]: addressed here https://github.com/GFNOrg/torchgfn/pull/286/files

[the remark this answers is not in the document]
Comment (saleml) [reply src/gfn/gflownet/detailed_balance.py:65]: Thanks, nice yolo
